### DISCUSSION TITLE
Recoding of android gradle for better functionality

### DISCUSSION THREAD

**Discussion by IsaacPere** [ionic-team/capacitor#6609] [Ideas]
Hello Community 

I am having a problem in running the capacitor app to android physical device and after of investigating. I have find out that the problem is the android gradle and its structures and synchronization to the build, variable and settings gradle files. And so can anyone who is proficiency in gradle language recode the android gradle folder and its associated files and be comparing the different mobile cross platform that are in the development aspects, I truly believe that flutter framework provide the best inspiration for gradle restructure as it can be able to install and run perfectly in android physical device through its command line interface.

Also be recoding the android gradle it will provide the best way for more android app developers to jump in easily and make app easily to all who will to venture to this marvelous world of providing people with solutions in their mobile phone and hence making trade and idea exploration easy and fast.

What I am hoping is that be recoding the gradle and its associated files it will help the community focus on how to make the framework better be making and deploying plugins that would make it easy for ideas to become life.
Thank You in Advance for anyone who will take up this initiative and I will be available for anyone who will want to to beta test the recode gradle for options, suggestions and many idea features to add to the code.

**Comment by jcesarmobile** [comment]
Not sure what you mean, capacitor uses standard gradle, the gradle folder contains the gradle wrapper that is downloaded with `gradlew wrapper --distribution-type all --gradle-version 8.0.2 --warning-mode all` command.

What errors are you getting and what commands are you running for getting those errors?

**Reply by IsaacPere** [reply]
Hello I have created a GitHub repository under the name CapacitorPWA, please clone it and try to run and then you will see where the problems is and also I will share some screenshots of the project.
I will be curious to hear from you at the soonest about the cause of the issues and if there will be a resolution 
Thank you
![Screenshot_2023-05-22_20_05_21](https://github.com/ionic-team/capacitor/assets/73386133/09ddbc6e-53ae-4fac-b3c9-8cfba0ff8c3d)

**Reply by jcesarmobile** [reply]
The error says you need an `ANDROID_HOME` environment variable pointing to your Android SDK folder.
If flutter doesn't need it to be set, it probably scans your computer to find it.

**Reply by IsaacPere** [reply]
Okay what about the configuration of the app and capacitor-cordova-plugins folder 
How can we solve it?

**Reply by jcesarmobile** [reply]
Those are just warnings, flatDir is used for Cordova plugins that have aar files. That doesn't prevent the app from building.

**Reply by IsaacPere** [reply]
I think i have forgot how to setting android_home environment variable in Kali Linux can you help

**Reply by IsaacPere** [reply]
> The error says you need an `ANDROID_HOME` environment variable pointing to your Android SDK folder. If flutter doesn't need it to be set, it probably scans your computer to find it.

I think if you set it like flutter framework it will make it easy for building process to be easier and faster
What do you think?

**Comment by IsaacPere** [comment]
Help me in the final steps of running the applications and please fell free to use Debian version of Linux because I have set android home environment variable but nothing is happening during running it to physical device.  
I fell that that I am missing something important 
Screenshots are below.
![Screenshot_2023-05-23_20_51_07](https://github.com/ionic-team/capacitor/assets/73386133/780f6e90-2e18-4072-a4a2-c991855774dc)
![Screenshot_2023-05-23_20_52_08](https://github.com/ionic-team/capacitor/assets/73386133/516c4c45-ef03-416d-84d1-77c85d24e5e0)

**Comment by IsaacPere** [comment]
Hello,
Good News I have been able to run the application through physical device and I noticed that all that I had to do is to specify the Android Home android environment variable in local.properties and ./gradlew cleanBuild in the android folder and all was okay.
thanks for everything.
